reference/reference.go

```diff
@@ -236,6 +237,18 @@ func WithDigest(name Named, digest digest.Digest) (Canonical, error) {
 	}, nil
 }
 
+// Match reports whether ref matches the specified pattern.
+func Match(pattern string, ref Reference) bool {
+	matched, err := path.Match(pattern, ref.String())
+	if err != nil {
+		return false
+	}
+	if namedRef, isNamed := ref.(Named); isNamed && !matched {
+		matched, _ = path.Match(pattern, namedRef.Name())
```

Comment:
I'm confused about why `Match` is called twice. Seems like this is returning `matches(ref.String()) || matches(ref.Name())`. Why isn't `ref.String()` enough?

Also, the eror is not checked here.